### PR TITLE
RT-1.30: Added deviation bgp_rib_oc_path_unsupported and updated script to delete the policy applied in bgp

### PR DESCRIPTION
**Pull request by mohanasm**
1. Script is currently replacing the policy without deleting the policy reference in import/export in BGP. Updated script to delete the policy-reference while deleting the routing-policy.
2. Also added deviation bgp_rib_oc_path_unsupported.

### CODE REVIEW

File: feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go [hunk 1]
```diff
@@ -234,146 +235,220 @@ func TestBGPNestedPolicies(t *testing.T) {
...
+func configureImportRoutingPolicy(t *testing.T, dut *ondatra.DUTDevice, operation string) {
 	batch := &gnmi.SetBatch{}
 	root := &oc.Root{}
 	rp := root.GetOrCreateRoutingPolicy()
 
-	// Configure a route-policy to set the local preference.
-	pdef1 := rp.GetOrCreatePolicyDefinition(v4LPPolicy)
-	stmt1, err := pdef1.AppendNewStatement(v4LPStatement)
+	// Configure PERMIT-ALL policy
+	pdef := rp.GetOrCreatePolicyDefinition("PERMIT-ALL")
```
Comment (nsadhasivam) [inline feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go:244]: Please have the policy name "PERMIT-ALL" defined as a const 
same for newstatement 20

File: feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go [hunk 2]
```diff
@@ -234,146 +235,220 @@ func TestBGPNestedPolicies(t *testing.T) {
...
+		prefixSet.GetOrCreatePrefix(advertisedIPv41.cidr(t), maskLenExact)
 
-	if !deviations.SkipSetRpMatchSetOptions(dut) {
-		stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetMatchSetOptions(oc.RoutingPolicy_MatchSetOptionsRestrictedType_ANY)
+		if !deviations.SkipSetRpMatchSetOptions(dut) {
+			stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetMatchSetOptions(oc.RoutingPolicy_MatchSetOptionsRestrictedType_ANY)
+		}
+		stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetPrefixSet(v4PrefixSet)
+	}
+	if operation == "set" {
```
Comment (nsadhasivam) [inline feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go:285]: Why do we need multiple if else for operation "set" and "delete".
Can we club them up with the if operation == "set" at line 251
	
	if operation == "set" {
		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
	} else if operation == "delete" {
		gnmi.BatchDelete(batch, gnmi.OC().RoutingPolicy().Config())
		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
	}
	dni := deviations.DefaultNetworkInstance(dut)
	if operation == "set" {
		// Configure the nested policy.
		rpPolicy := root.GetOrCreateRoutingPolicy()
		statPath := rpPolicy.GetOrCreatePolicyDefinition(v4LPPolicy).GetStatement(v4LPStatement).GetOrCreateConditions()
		statPath.SetCallPolicy(v4PrefixPolicy)
		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rpPolicy)
	} else if operation == "delete" {
		gnmi.BatchDelete(batch, gnmi.OC().RoutingPolicy().Config())
		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
	}

File: feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go [hunk 3]
```diff
@@ -234,146 +235,220 @@ func TestBGPNestedPolicies(t *testing.T) {
...
+		gnmi.BatchDelete(batch, gnmi.OC().RoutingPolicy().Config())
+		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
+	}
 	// Configure the parent BGP import policy.
-	path := gnmi.OC().NetworkInstance(dni).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).Bgp().Neighbor(atePort1.IPv4).AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).ApplyPolicy()
-	policy := root.GetOrCreateNetworkInstance(dni).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).GetOrCreateBgp().GetOrCreateNeighbor(atePort1.IPv4).GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).GetOrCreateApplyPolicy()
-	policy.SetImportPolicy([]string{v4LPPolicy})
-	if !deviations.RoutingPolicyChainingUnsupported(dut) {
-		gnmi.Update(t, dut, path.Config(), policy)
+	if deviations.RoutePolicyUnderAFIUnsupported(dut) {
```
Comment (nsadhasivam) [inline feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go:303]: Multiple time if operation == "set" / if operation == "delete"  is used
	Please clean it up through the entire changes
	
	if deviations.RoutePolicyUnderAFIUnsupported(dut) {
		//policy under peer group
		path := gnmi.OC().NetworkInstance(dni).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).Bgp().PeerGroup(peerGrpNamev4).ApplyPolicy()
		policy := root.GetOrCreateNetworkInstance(dni).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).GetOrCreateBgp().GetOrCreatePeerGroup(peerGrpNamev4).GetOrCreateApplyPolicy()
		policy.SetImportPolicy([]string{v4LPPolicy})
		if operation == "set" {
			gnmi.BatchReplace(batch, path.Config(), policy)
		} else if operation == "delete" {
			gnmi.BatchDelete(batch, path.Config())
		}
	} else {
		path := gnmi.OC().NetworkInstance(dni).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).Bgp().Neighbor(atePort1.IPv4).AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).ApplyPolicy()
		policy := root.GetOrCreateNetworkInstance(dni).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).GetOrCreateBgp().GetOrCreateNeighbor(atePort1.IPv4).GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).GetOrCreateApplyPolicy()
		policy.SetImportPolicy([]string{v4LPPolicy})
		if operation == "set" {
			if !deviations.RoutingPolicyChainingUnsupported(dut) {
				gnmi.BatchUpdate(batch, path.Config(), policy)
			} else {
				gnmi.BatchReplace(batch, path.Config(), policy)
			}
		} else if operation == "delete" {
			gnmi.BatchDelete(batch, path.Config())
		}
	}

**Review by nsadhasivam — CHANGES_REQUESTED** [review]

**Comment by self-maurya** [comment]
If I understand correctly, we are dropping the polices at the end of each subtest.

How about we drop the policies at the start of each subtest without any condition, then run the subtest without any "set"/"delete" operation? Please help me understand if I'm missing something here.

File: feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go (hunk 1 above)
Comment (mohanasm) [reply feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go]: Updated. Please check.

File: feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go (hunk 2 above)
Comment (mohanasm) [reply feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go]: Updated the code to remove multiple if else condition.

File: feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go (hunk 3 above)
Comment (mohanasm) [reply feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go:295]: Updated the code to reduce multiple if else condition.

**Comment by mohanasm** [comment]
> @mohanasm If I understand correctly, we are dropping the polices at the end of each subtest.
> 
> How about we drop the policies at the start of each subtest without any condition, then run the subtest without any "set"/"delete" operation? Please help me understand if I'm missing something here.

Included code to set and delete configurations for each subtest case to ensure that only the necessary configurations are applied for each sub-test case without causing conflicts. This led to multiple if-else conditions, so the code has been modified to remove the previous subtest configuration at the start of each sub-test case.

**Review by nsadhasivam — APPROVED** [review]
LGTM!

**Pull request merged**